### PR TITLE
Comment refactor Part 1

### PR DESCRIPTION
**Pull request by jarvislin**
Currently, there are two entry points to the screen of Comments Details:
1. Notifications List -> Comment
2. My Site -> Comments List -> Comment

For some reason, we implemented every thing in that big `CommentDetailFragment`. There is some logic that is only triggered by specific entry points. To maintain the simplicity of the parent class, I suggest relocating this code to a subclass. Furthermore, converting subclasses to Kotlin can enhance development efficiency.

This a PR contains:

1. Create two Fragments (which extend `CommentDetailFragment`) for different purpose.
2. Convert a file to Kotlin.
3. Remove some redundant logic

There will be follow-up PRs for furthering refactors.

-----

## To Test:

1. Sign in JP app
2. Go to the Notifications tab
3. Click on `Comment` type of notification
4. Every thing should be the same as on the `feature/notifications_refresh_p2` branch.
5. Go to My Site -> Comments, click on a comment
7. Every thing should be the same as on the `feature/notifications_refresh_p2` branch.
8. Done, thank you.

-----

## Regression Notes

1. Potential unintended areas of impact

    - notifications, comment

11. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

12. What automated tests I added (or what prevented me from doing so)

    - none

-----

## PR Submission Checklist:

Skipped

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

Skipped

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.kt [hunk 1]
```diff
@@ -385,9 +383,6 @@ class NotificationsListFragment : Fragment(R.layout.notifications_list_fragment)
             }
             val detailIntent = getOpenNoteIntent(activity, noteId)
             detailIntent.putExtra(NOTE_INSTANT_REPLY_EXTRA, shouldShowKeyboard)
-            if (!TextUtils.isEmpty(replyText)) {
-                detailIntent.putExtra(NOTE_PREFILLED_REPLY_EXTRA, replyText)
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.kt:389]: I think we can also remove the `NOTE_PREFILLED_REPLY_EXTRA` declaration above.

**Review by antonis — APPROVED** [review]
Thank you for this refactoring @jarvislin 🙇 
The code changes LGTM and the app behaved as expected in my tests.
I've only left a comment for a redundant constant declaration.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.kt (hunk 1 above)
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsListFragment.kt:389]: done

**Pull request merged**